keps/3125-maximum-execution-time/README.md

```diff
@@ -0,0 +1,183 @@
+# KEP-3125: Workload Maximum Execution Time
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Constants](#constants)
+  - [Controller](#controller)
+    - [Workload](#workload)
+    - [Jobs / Jobframework](#jobs--jobframework)
+  - [Webhooks](#webhooks)
+    - [Workload](#workload-1)
+    - [Jobs](#jobs)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+      - [controller/core/workload](#controllercoreworkload)
+      - [controller/jobs/job](#controllerjobsjob)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Add the ability to set a maximum execution time of a Job.
+
+## Motivation
+
+Different Jobs CRDs have a field with similar semantics, but there is no standard. For example `batch/Job` has `spec.activeDeadlneSeconds`, while JobSet does not have such an API for now.
+
+
+### Goals
+
+We would like to have a common API of setting and enforcing the maximum execution time of a Job.
+
+
+### Non-Goals
+
+Change the behavior of the Kueue's scheduler based on the maximum execution time.
+
+## Proposal
+
+A new Kueue specific label is added `kueue.x-k8s.io/max-exec-time-seconds` containing
+the maximum number of seconds the Job is expected to execute.
+
+The duration is passed to the Job's Workload.
+
+If a workload stays admitted for longer that it is expected it will be Deactivated. 
+
+### User Stories (Optional)
+
+#### Story 1
+
+**TBD**
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+We can accumulate the time a Workload has spent in Admitted state in previous Admit - Evict cycles, however:
```

Comment:
> For 1: the deletion of workload could also affect other features such as waitForPodsReady.

Just because other feature are affected it's not an indication that a concern is irrelevant.

> For 2: most AI/ML Jobs would have check-pointing mechanisms to not reset progress on eviction.

Even with a check-pointing mechanism an `Evict - Admit` cycle is adding an overhead and the number of such cycles is very likely unknown to the job submitter.

Given this and the fact that accounting for the previously spent time as Admitted requires additional API changes that will be hard to revert, I  believe it's better to skip the cumulative approach in the initial implementation, and potentially do it in a follow-up.

Reply:
Not an expert but it seems like what @mimowo is requesting makes sense. @mwielgus WDYT?

Reply:
I agree with @mimowo.

Reply:
done